### ISSUE TITLE
NYC or spawn-wrap bug?

### ISSUE THREAD

**Issue by jamestalmage** [jamestalmage/__nyc-ipc-argv-bug#1]
I suspect `spawn-wrap`.

// @isaacs @bcoe
// @novemberborn @develar

**Comment by addaleax** [comment]
fwiw, adding message forwarding to `foreground-child` fixes it (because that’s used by `spawn-wrap` for passing `execArgv`):

``` diff
diff --git a/index.js b/index.js
index 21a395a64ceb..8e548b63ada2 100644
--- a/index.js
+++ b/index.js
@@ -60,8 +60,14 @@ module.exports = function (program, args, cb) {
     args = [].slice.call(arguments, 1, arrayIndex)
   }

+  var spawnOpts = { stdio: ['inherit', 'inherit', 'inherit'] }
+
+  if (process.send) {
+    spawnOpts.stdio.push('ipc')
+  }
+
   var spawnfn = needsCrossSpawn(program) ? crossSpawn : spawn
-  var child = spawnfn(program, args, { stdio: 'inherit' })
+  var child = spawnfn(program, args, spawnOpts)

   var childExited = false
   signalExit(function (code, signal) {
@@ -89,5 +95,15 @@ module.exports = function (program, args, cb) {
     })
   })

+  if (process.send) {
+    child.on('message', function (message, sendHandle) {
+      process.send(message, sendHandle)
+    })
+
+    process.on('message', function (message, sendHandle) {
+      child.send(message, sendHandle)
+    })
+  }
+
   return child
 }
```

**Comment by jamestalmage** [comment]
Can you open a PR with those changes?

Also, is it necessary to declare `ipc` in the 4th slot? Could you not just use `inherit`, or `4` and then avoid forwarding messages?

**Comment by addaleax** [comment]
> Can you open a PR with those changes?

I’ll do, but I’m not sure it’s actually the right place to fix it.

> Also, is it necessary to declare `ipc` in the 4th slot? Could you not just use `inherit`, or `4` and then avoid forwarding messages?

Hmm… I’m pretty sure there’s no guarantee or reasonable way to estimate which FD the IPC channel will take on, so it would probably be trickier than that.

**Comment by jamestalmage** [comment]
> I’m not sure it’s actually the right place to fix it.

Your solution seems reasonable to me.

**Comment by bcoe** [comment]
@jamestalmage this should now be released in `nyc@6.6.0`.

mind giving it a spin on a few of your repos?

**Comment by jamestalmage** [comment]
6.0.0 fixed this repo
